nemo/collections/multimodal/speech_llm/data/lhotse_dataset.py

```diff
@@ -0,0 +1,493 @@
+import copy
+import random
+from typing import Optional
+
+import numpy as np
+import torch.utils.data
+from lhotse.dataset.collation import collate_vectors as collate_vectors_lhotse
+from nemo.utils import logging
+
+
+def collate_vectors(items, max_length: int, padding_value):
+    vectors = collate_vectors_lhotse(items, padding_value=padding_value)
+    if max_length > vectors.size(1):
+        vectors = torch.cat(
+            [vectors, padding_value * torch.ones(vectors.size(0), max_length - vectors.size(1), dtype=vectors.dtype)],
+            dim=1,
+        )
+    if items[0].shape[0] < 1:
+        vectors = vectors.long()
+    return vectors
+
+
+def ceil_to_nearest(n, m):
+    return (n + m - 1) // m * m
+
+
+class TextProcessing:
```

Comment:
This class needs more documentation on what is it doing, how to use its API, and what are the expected input and output formats. Also, it only has private methods right now, the main API method should be public (no underscore at the beginning).

I'd expect a docstring of kind: ```this class is used to convert X to Y. in order to do so, it performs A, B, C, and D. the expect format of X is .... the expected format of Y is ...```

since it's used to convert text to prompts to token ids, I'd like to see full documentation of the prompt template/schema

Reply:
Done